lib/internal/source_map/prepare_stack_trace.js

```diff
@@ -178,28 +168,46 @@ function getErrorSource(
 
 function getOriginalSource(payload, originalSourcePath) {
   let source;
-  const originalSourcePathNoScheme =
-    StringPrototypeStartsWith(originalSourcePath, 'file://') ?
-      fileURLToPath(originalSourcePath) : originalSourcePath;
+  // payload.sources has been normalized to be an array of absolute urls.
   const sourceContentIndex =
     ArrayPrototypeIndexOf(payload.sources, originalSourcePath);
   if (payload.sourcesContent?.[sourceContentIndex]) {
     // First we check if the original source content was provided in the
     // source map itself:
     source = payload.sourcesContent[sourceContentIndex];
-  } else {
+  } else if (StringPrototypeStartsWith(originalSourcePath, 'file://')) {
     // If no sourcesContent was found, attempt to load the original source
     // from disk:
+    debug(`read source of ${originalSourcePath} from filesystem`);
+    const originalSourcePathNoScheme = fileURLToPath(originalSourcePath);
     try {
       source = readFileSync(originalSourcePathNoScheme, 'utf8');
     } catch (err) {
       debug(err);
       source = '';
     }
+  } else {
+    source = '';
   }
   return source;
 }
 
+function getSourceMapErrorSource(fileName, lineNumber, columnNumber) {
+  const sm = findSourceMap(fileName);
+  if (sm == null) {
```

Comment:
nit: I think this could be `if(sm === undefined)`? Is there a case where it would ever be `null` or `falsy`?